ema_workbench/em_framework/optimization.py

```diff
@@ -512,28 +667,72 @@ class ArchiveLogger(AbstractConvergenceMetric):
     base_filename : str, optional
 
 
-    TODO:: put it in a tarbal instead of dedicated directory
-
     """
 
-    def __init__(self, directory, decision_varnames, outcome_varnames, base_filename="archive"):
+    def __init__(
+        self, directory, decision_varnames, outcome_varnames, base_filename="archives.tar.gz"
+    ):
         super().__init__("archive_logger")
+
+        # FIXME how to handle case where directory already exists
         self.directory = os.path.abspath(directory)
+        self.temp = os.path.join(self.directory, "tmp")
+        os.mkdir(self.temp)
+
         self.base = base_filename
         self.decision_varnames = decision_varnames
         self.outcome_varnames = outcome_varnames
-        self.index = 0
+        self.tarfilename = os.path.join(self.directory, base_filename)
+
+        # self.index = 0
 
     def __call__(self, optimizer):
-        self.index += 1
+        archive = to_dataframe(optimizer.result, self.decision_varnames, self.outcome_varnames)
+        archive.to_csv(os.path.join(self.temp, f"{optimizer.nfe}.csv"))
```

Comment:
What/why is the to_csv writing part for?

Reply:
I write each archive as a csv file to a temporary directory.

Next, upon completing the optimization, the entire temp directory is turned into a tarball and the temp directory is removed. 

 I like datafile to also be human readable, hence the use of a csv file. If you unzip the archive tarball, you see a bunch of csv files.

Reply:
Right we discussed that. Is CSV the best/fastest format for this, especially if it's only temporary? Could a Pickle be more suitable? (with [DataFrame.to_pickle](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_pickle.html) / [pandas.read_pickle](https://pandas.pydata.org/docs/reference/api/pandas.read_pickle.html))

Reply:
It is not a temporary format. The tarball does contain the csv files. 

When I first started with the workbench, I did use pickle for storing results. It is compact but not practical. Pickle files do not transfer well across machines or python versions. Moreover, it makes it impossible to analyze results in other contexts. I rather have slightly larger files that are human-readable and do transfer well from machine to machine.